cmake/multi_image.cmake

```diff
@@ -321,54 +321,65 @@ function(add_child_image_from_source)
     #       files must be used instead of the child image default configs.
     #       The append a child image default config, place the additional settings
     #       in `child_image/<ACI_NAME>.conf`.
-    set(ACI_CONF_DIR ${APPLICATION_CONFIG_DIR}/child_image)
-    set(ACI_NAME_CONF_DIR ${APPLICATION_CONFIG_DIR}/child_image/${ACI_NAME})
-    if (NOT ${ACI_NAME}_CONF_FILE)
-      ncs_file(CONF_FILES ${ACI_NAME_CONF_DIR}
-        BOARD ${ACI_BOARD}
-        # Child image always uses the same revision as parent board.
-        BOARD_REVISION ${BOARD_REVISION}
-        KCONF ${ACI_NAME}_CONF_FILE
-        DTS ${ACI_NAME}_DTC_OVERLAY_FILE
-        BUILD ${CONF_FILE_BUILD_TYPE}
-        )
-      # Place the result in the CMake cache and remove local scoped variable.
-      foreach(file CONF_FILE DTC_OVERLAY_FILE)
-        if(DEFINED ${ACI_NAME}_${file})
-          zephyr_file_suffix(${ACI_NAME}_${file} SUFFIX ${${ACI_NAME}_FILE_SUFFIX})
-
-          set(${ACI_NAME}_${file} ${${ACI_NAME}_${file}} CACHE STRING
-            "Default ${ACI_NAME} configuration file" FORCE
-            )
-	  set(${ACI_NAME}_${file})
+    zephyr_get(COMMON_CHILD_IMAGE_CONFIG_DIR)
+    string(CONFIGURE "${COMMON_CHILD_IMAGE_CONFIG_DIR}" COMMON_CHILD_IMAGE_CONFIG_DIR)
+    foreach(config_dir ${APPLICATION_CONFIG_DIR} ${COMMON_CHILD_IMAGE_CONFIG_DIR} )
+      set(ACI_CONF_DIR ${config_dir}/child_image)
+      set(ACI_NAME_CONF_DIR ${config_dir}/child_image/${ACI_NAME})
+      if (NOT ${ACI_NAME}_CONF_FILE)
+        ncs_file(CONF_FILES ${ACI_NAME_CONF_DIR}
+          BOARD ${ACI_BOARD}
+          # Child image always uses the same revision as parent board.
+          BOARD_REVISION ${BOARD_REVISION}
+          KCONF ${ACI_NAME}_CONF_FILE
+          DTS ${ACI_NAME}_DTC_OVERLAY_FILE
+          BUILD ${CONF_FILE_BUILD_TYPE}
+          SUFFIX ${${ACI_NAME}_FILE_SUFFIX}
```

Comment:
These two options are exclusive, if there is a suffix then it needs to override the build type, it should not append build type and suffix which seems to be the case:
```
cmake -GNinja -DBOARD=nrf52840dk/nrf52840 -DCONF_FILE=prj_base.conf -Dmcuboot_FILE_SUFFIX=base -DCONFIG_BOOTLOADER_MCUBOOT=y ..

-- Found BOARD.dts: /tmp/bb/zephyr/boards/nordic/nrf52840dk/nrf52840dk_nrf52840.dts
-- Found devicetree overlay: /tmp/bb/zephyr/samples/hello_world/child_image/mcuboot/boards/nrf52840dk_nrf52840_base_base.overlay
-- Found devicetree overlay: /tmp/bb/nrf/modules/mcuboot/usb.overlay
-- Generated zephyr.dts: /tmp/bb/zephyr/samples/hello_world/_AA/mcuboot/zephyr/zephyr.dts
-- Generated devicetree_generated.h: /tmp/bb/zephyr/samples/hello_world/_AA/mcuboot/zephyr/include/generated/devicetree_generated.h
-- Including generated dts.cmake file: /tmp/bb/zephyr/samples/hello_world/_AA/mcuboot/zephyr/dts.cmake

warning: FLASH_MAP (defined at subsys/storage/flash_map/Kconfig:10) has direct dependencies FLASH_HAS_DRIVER_ENABLED with value n, but is currently being y-selected by the following symbols:
 - MCUBOOT_DEVICE_SETTINGS (defined at /tmp/bb/bootloader/mcuboot/boot/zephyr/Kconfig:707), with value y, direct dependencies y (value: y)

error: Aborting due to Kconfig warnings

Parsing /tmp/bb/bootloader/mcuboot/boot/zephyr/Kconfig
Loaded configuration '/tmp/bb/zephyr/boards/nordic/nrf52840dk/nrf52840dk_nrf52840_defconfig'
Merged configuration '/tmp/bb/zephyr/samples/hello_world/child_image/mcuboot/prj_base_base.conf'
Merged configuration '/tmp/bb/zephyr/samples/hello_world/child_image/mcuboot/boards/nrf52840dk_nrf52840_base_base.conf'
Merged configuration '/tmp/bb/nrf/subsys/partition_manager/partition_manager_enabled.conf'
Merged configuration '/tmp/bb/zephyr/samples/hello_world/_AA/mcuboot/zephyr/misc/generated/extra_kconfig_options.conf'
```

Reply:
this is an upstream issue and should be fixed there.

Reply:
Cannot replicate in upstream, feature works fine